### ISSUE TITLE
openr,/compress and assoc() is broken 

### ISSUE THREAD

**Issue by slayoo** [gnudatalanguage/gdl#137]
As reported by @opoplawski on SF.net (https://sourceforge.net/p/gnudatalanguage/bugs/632/): 

Here's a full test showing the problem:

```
pro TEST_ASSOC
;
;
file='test_zip.gz'
;
txt='"GDL - GNU Data Language\nA free IDL (Interactive Data Language) compatible incremental\n'
txt=txt+'compiler (i. e. runs IDL programs)."'
;
SPAWN, 'echo -e '+txt+' | gzip > '+file
OPENR, fd, file, /get, /compress, /delete
;
; Assoc test
POINT_LUN, fd, 0
a=ASSOC(fd, BYTARR(5))

print,"a[0]=",string(a[0])
print,"a[0]=",string(a[0])
print,"a[1]=",string(a[1])
print,"a[1]=",string(a[1])
print,"a[1]=",string(a[1])
print,"a[2]=",string(a[2])
;
CLOSE, fd
FREE_LUN, fd
;
file='test_assoc.dat'
;
txt='"GDL - GNU Data Language\nA free IDL (Interactive Data Language) compatible incremental\n'
txt=txt+'compiler (i. e. runs IDL programs)."'
;
SPAWN, 'echo -e '+txt+' > '+file
OPENR, fd, file, /get, /delete
;
; Assoc test
POINT_LUN, fd, 0
a=ASSOC(fd, BYTARR(5))

print,"a[0]=",string(a[0])
print,"a[0]=",string(a[0])
print,"a[1]=",string(a[1])
print,"a[1]=",string(a[1])
print,"a[1]=",string(a[1])
print,"a[2]=",string(a[2])
;
CLOSE, fd
FREE_LUN, fd
;
end
```

Results in:

```
GDL> test_assoc
a[0]=GDL -
a[0]=GDL -
a[1]= GNU
a[1]=Data
a[1]=Langu
a[2]=age
A
a[0]=GDL -
a[0]=GDL -
a[1]= GNU
a[1]= GNU
a[1]= GNU
a[2]=Data
```

So for /compress, when n!=0 all subsequent accesses on the array increment the pointer along the file.

**Comment by slayoo** [comment]
With 0.9.8 from the Debian package:
```
GDL> test_assoc
% Compiled module: TEST_ASSOC.
a[0]=GDL -
a[0]=GDL -
a[1]= GNU 
a[1]= GNU 
a[1]= GNU 
a[2]=Data 
a[0]=GDL -
a[0]=GDL -
a[1]= GNU 
a[1]= GNU 
a[1]= GNU 
a[2]=Data 
```

@opoplawski please confirm if there were any news on it in the meantime or which platform to look for to reproduce it?

**Comment by opoplawski** [comment]
It does appear to be fixed for in 0.9.8.  This was on EL7 x86_64.
